### ISSUE TITLE
perl-5.8.3 on Linux

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#7034]
Migrated from [rt.perl.org#24890](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=24890) (status was 'resolved')

Searchable as RT24890$

**Comment by p5pRT** [comment]
### From warren.dodge@tek.com

#### Created by warren.dodge@tek.com




I just built perl\-5\.8\.3\-RC1 for Linux

Linux tekcs22 2\.4\.21\-4\.ELhugemem \#1 SMP Fri Oct 3 17&#8203;:31&#8203;:20 EDT 2003 i686 i686 i386 GNU/Linux

It went well except I have one area of questions\. I don't use the dbm stuff
so I just want to make sure what is is doing makes sense to you\.


On this version of Linux there is no \-lndbm available\. In the library list
presented early in the configure there was not a ndbm entry\.

What libraries to use?
\[\-lnsl \-lgdbm \-ldb \-ldl \-lm \-lcrypt \-lutil \-lpthread \-lc\] 

However later in the Configure where it asks

What extensions do you wish to load dynamically?

That list has NDBM\_File in it\. It seems like it should not be there\.
On the next question about what to load statically I answered "none" where is
presented NDBM\_File as the default answer\.

Then in the make test the following was shown which looks like it did fine\.

ext/NDBM\_File/t/ndbm\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.skipping test on this platform
ext/ODBM\_File/t/odbm\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.skipping test on this platform
ext/SDBM\_File/t/sdbm\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.ok
lib/AnyDBM\_File\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.ok
lib/Memoize/t/tie\_gdbm\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.ok
lib/Memoize/t/tie\_ndbm\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.skipping test on this platform
lib/Memoize/t/tie\_sdbm\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.ok

Many of the gcc commands used \-I/usr/include/gdbm which has the following
files

ll /usr/include/gdbm
total 28
drwxr\-xr\-x    2 root         4096 Jan  5 12&#8203;:56 \./
drwxr\-xr\-x  114 root         8192 Jan  8 23&#8203;:10 \.\./
\-rw\-r\-\-r\-\-    1 root         1563 Jan 25  2003 dbm\.h
\-rw\-r\-\-r\-\-    1 root         4744 Jan 25  2003 gdbm\.h
\-rw\-r\-\-r\-\-    1 root         2003 Jan 25  2003 ndbm\.h

I am not sure why the ndbm\.h or dbm\.h is there and if it confuses anything\. 

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=low

Site configuration information for perl v5.8.3:

Configured by warrend at Mon Jan 12 15:38:25 PST 2004.

Summary of my perl5 (revision 5.0 version 8 subversion 3) configuration:
  Platform:
    osname=linux, osvers=2.4.21-4.elhugemem, archname=i686-linux-thread-multi
    uname='linux tekcs22 2.4.21-4.elhugemem #1 smp fri oct 3 17:31:20 edt 2003 i686 i686 i386 gnulinux '
    config_args=''
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=define use5005threads=undef useithreads=define usemultiplicity=define
    useperlio=define d_sfio=undef uselargefiles=define usesocks=undef
    use64bitint=undef use64bitall=undef uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='gcc', ccflags ='-D_REENTRANT -D_GNU_SOURCE -DTHREADS_HAVE_PIDS -fno-strict-aliasing -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm',
    optimize='-O3',
    cppflags='-D_REENTRANT -D_GNU_SOURCE -DTHREADS_HAVE_PIDS -fno-strict-aliasing -I/usr/include/gdbm'
    ccversion='', gccversion='3.2.3 20030502 (Red Hat Linux 3.2.3-20)', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=4, prototype=define
  Linker and Libraries:
    ld='gcc', ldflags =''
    libpth=/lib /usr/lib
    libs=-lnsl -lgdbm -ldb -ldl -lm -lcrypt -lutil -lpthread -lc
    perllibs=-lnsl -ldl -lm -lcrypt -lutil -lpthread -lc
    libc=/lib/libc-2.3.2.so, so=so, useshrplib=true, libperl=libperl5.8.3-RC1.so
    gnulibc_version='2.3.2'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-rdynamic -Wl,-rpath,/tools/perl5.8.2/lib/5.8.3/i686-linux-thread-multi/CORE'
    cccdlflags='-fpic', lddlflags='-shared'

Locally applied patches:
    RC1


@INC for perl v5.8.3:
    /tools/perl5.8.2/lib/5.8.3/i686-linux-thread-multi
    /tools/perl5.8.2/lib/5.8.3
    /tools/perl5.8.2/lib/site_perl/5.8.3/i686-linux-thread-multi
    /tools/perl5.8.2/lib/site_perl/5.8.3
    /tools/perl5.8.2/lib/site_perl
    .


Environment for perl v5.8.3:
    HOME=/login/warrend/lxhome
    LANG=en_US.UTF-8
    LANGUAGE (unset)
    LANGVAR=en_US.UTF-8
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/usr/bin:/bin:/usr/X11R6/bin
    PERL_BADLANG (unset)
    SHELL=/bin/tcsh

```
</details>

**Comment by p5pRT** [comment]
### From @doughera88

On Mon\, 13 Jan 2004\, Warren\.Dodge@&#8203;Tek\.Com wrote&#8203;:

> I just built perl\-5\.8\.3\-RC1 for Linux

> It went well except I have one area of questions\. I don't use the dbm stuff
> so I just want to make sure what is is doing makes sense to you\.

> On this version of Linux there is no \-lndbm available\. In the library list
> presented early in the configure there was not a ndbm entry\.
>
> What libraries to use?
> \[\-lnsl \-lgdbm \-ldb \-ldl \-lm \-lcrypt \-lutil \-lpthread \-lc\]

Ok\.  This is correct\.

> However later in the Configure where it asks
>
> What extensions do you wish to load dynamically?
>
> That list has NDBM\_File in it\. It seems like it should not be there\.

Configure only puts NDBM\_File in that list if Configure determines that it
can safely \#include \<ndbm\.h> and also if it thinks it can load the
function dbm\_open\(\) from somewhere in the library list\.

I expect that on your system the dbm\_open\(\) function might be contained in
\-lgdbm as part of some compatibility layer or some such\.


> On the next question about what to load statically I answered "none" where is
> presented NDBM\_File as the default answer\.

That's ok too\.  For static extensions\, Configure suggests all the
available extensions that aren't already included as dynamic extensions\.
That is\, Configure thinks NDBM\_File will work for you\.  You chose not to
use it as a dynamic extension\.  However\, Configure is allowing for the
possibility that you want it as a static extension\.  Omitting it is\, of
course\, just fine\.

>
> Then in the make test the following was shown which looks like it did fine\.
>
> ext/NDBM\_File/t/ndbm\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.skipping test on this platform

Yup\, that's fine \-\-you chose to skip it\.

> Many of the gcc commands used \-I/usr/include/gdbm which has the following
> files
> \-rw\-r\-\-r\-\-    1 root         2003 Jan 25  2003 ndbm\.h

Yup\.  Some distributions stick the \*dbm\.h headers there\.  Configure knows
about that and knows how to find them\.

In short\, your set up is fine\.  Your system probably does have some
sort of ndbm compatibility available; that's why Configure suggested it\.
Keeping it would have been fine\.  Skipping it as you did is fine too\.

Hope this clears it up\,

\-\- 
&nbsp;    Andy Dougherty        doughera@&#8203;lafayette\.edu

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From warren.dodge@tektronix.com



I have since determined how to make NDBM and ODBM work on linux\. It seems the
installation we have does not include the libgdbm\_compat library\. Not sure if
this is an oversite on Redhat's part or if they meant to\.

This library includes the dbmopen and other "obsolete" \( I assume\) functions\.

Once I locally installed gdbm with that library\, perl worked just fine with
the default answers\.

This also requires a tweak to the linux hints files for ODBM and NDBM to
include the libgdbm\_compat library\.


Thanks for all the help\.

**Comment by p5pRT** [comment]
@iabyn - Status changed from 'open' to 'resolved'
